### PR TITLE
508 implement file store backend

### PR DESCRIPTION
**Pull request by hdg4jivs**
## Describe your changes briefly
- A CRUD as described in https://github.com/DataMigration-International-AG/jivs-occ/issues/508 
- Add tests to test the code

## Issue ticket number with link
closes  /DataMigration-International-AG/jivs-occ#508
## Database changes
- Add schema jds
- Add table JIVS-DOCSTORE
- Add table JIVS-DOCUMENT-PARAM
- 
## Pre-Review checklist
- [x] I reviewed my code
- [x] The code is runnable
- [ ] I tested the code with edge cases
- [x] I wrote tests for my code

Please review. Thank you.

### CODE REVIEW

**Comment by hdg4jivs** [comment]
@mhn4jivs @Involute1 
The way I wrote the code for the service component is that it returns an Optional<SOMETHING> and not the SOMETHING. 
The controller is then checking if the Optional<> isPresent() and decides if it should return the actual value or null / empty List.

Should the Service check if the Optional is present and do above mentioned logic?

File: src/main/java/com/datamigration/jds/controller/DocStoreManager.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+		return deleted;
+	}
+
+	private List<DocumentDTO> getParamsForList(Optional<List<DocumentDTO>> documentDTOs)
+		throws JPEPersistenceException {
+		List<DocumentDTO> documentDTOList = new ArrayList<>();
+		if (documentDTOs.isPresent()) {
+			for (DocumentDTO documentDTO : documentDTOs.get()) {
+				DocumentDTO documentDTOWithParams = getDocumentParams(documentDTO);
+				documentDTOList.add(documentDTOWithParams);
```
Comment (Involute1) [inline src/main/java/com/datamigration/jds/controller/DocStoreManager.java:177]: nitpick: you already give it an Optional of List<DocumentDTO> why do you parse it again?

File: src/main/java/com/datamigration/jds/controller/DocStoreManager.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+			}
+		}
+		return documentDTOList;
+	}
+
+	private DocumentDTO getDocumentParams(DocumentDTO documentDTO) throws JPEPersistenceException {
+		Optional<Map<String, String>> result = documentService.getParams(documentDTO.id());
```
Comment (Involute1) [inline src/main/java/com/datamigration/jds/controller/DocStoreManager.java:184]: nitpick: i dont like the name of the method. Also why are we parsing this again you already have a documentdto?

File: src/main/java/com/datamigration/jds/persistence/DatabaseManager.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	 * @throws JPEPersistenceException if there is an error while establishing the database connection
+	 */
+	public static Connection connect() throws JPEPersistenceException {
+		Connection result;
+		try {
+			if (databaseConfig.getDbUrl() != null && databaseConfig.getDbUsername() != null
+				&& databaseConfig.getDbPassword() != null) {
+				result = DriverManager.getConnection(databaseConfig.getDbUrl(), databaseConfig.getDbUsername(),
+					databaseConfig.getDbPassword());
+				logger.debug("Connected to {}", result.getMetaData().getURL());
```
Comment (Involute1) [inline src/main/java/com/datamigration/jds/persistence/DatabaseManager.java:60]: nitpick: use jivs config for this

File: src/main/java/com/datamigration/jds/persistence/IDao.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+	Optional<T> getById(U id) throws JPEPersistenceException;
+
+	/**
+	 * Get all entities from the database. Returns an empty list if no entry was found.
+	 *
+	 * @return a list of all entities
+	 * @throws JPEPersistenceException if an error occurs during fetch
+	 */
+	Optional<List<T>> getAllAsList() throws JPEPersistenceException;
```
Comment (Involute1) [inline src/main/java/com/datamigration/jds/persistence/IDao.java:25]: nitpick: in my opinon we dont need the optinal here, on error we return an empty list

**Comment by mhn4jivs** [comment]
> @mhn4jivs @Involute1 The way I wrote the code for the service component is that it returns an Optional< SOMETHING > and not the SOMETHING. The controller is then checking if the Optional<> isPresent() and decides if it should return the actual value or null / empty List.
> 
> Should the Service check if the Optional is present and do above mentioned logic?



> @mhn4jivs @Involute1 The way I wrote the code for the service component is that it returns an Optional< SOMETHING > and not the SOMETHING. The controller is then checking if the Optional<> isPresent() and decides if it should return the actual value or null / empty List.
> 
> Should the Service check if the Optional is present and do above mentioned logic?

Imo the Controller will decide what to do.

File: src/main/java/com/datamigration/jds/persistence/db/DbDao.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import java.sql.SQLException;
+
+public class DbDao implements IDbDao {
+
+	@Override
+	public void createJDSSchema() throws JPEPersistenceException {
+		try (Connection connection = connect(); PreparedStatement preparedStatement = connection.prepareStatement(
+			IDbSQLs.CREATE_JDS_SCHEMA_SQL)) {
+			preparedStatement.execute();
+		} catch (SQLException | JPEPersistenceException e) {
```
Comment (Involute1) [inline src/main/java/com/datamigration/jds/persistence/db/DbDao.java:18]: nitpick: where logging?

File: src/main/java/com/datamigration/jds/persistence/IDao.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+	 */
+	T insert(T t) throws JPEPersistenceException;
+
+	/**
+	 * Update the given entity in the database. Throws an exception if an error occurs.
+	 *
+	 * @param t the entity to update
+	 * @throws JPEPersistenceException if an error occurs during persisting
+	 */
+	void update(T t) throws JPEPersistenceException;
```
Comment (Involute1) [inline src/main/java/com/datamigration/jds/persistence/IDao.java:43]: nitpick: should also return updated object

File: src/main/java/com/datamigration/jds/persistence/docstore/DocumentDao.java [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+			}
+		} catch (SQLException e) {
+			throw new JPEPersistenceException(e, ErrorCode.DB_WRITE_ERROR);
+		}
+		return result;
+	}
+
+	@Override
+	public void update(DocumentDTO documentDTO) throws JPEPersistenceException {
+
```
Comment (Involute1) [inline src/main/java/com/datamigration/jds/persistence/docstore/DocumentDao.java:77]: issue: why is this empty?

File: src/main/java/com/datamigration/jds/persistence/docstore/IDocumentDao.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+import com.datamigration.jds.model.dto.DocumentDTO;
+import com.datamigration.jds.persistence.IDao;
+import com.datamigration.jds.util.exceptions.checked.JPEPersistenceException;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+public interface IDocumentDao extends IDao<DocumentDTO, UUID> {
```
Comment (Involute1) [inline src/main/java/com/datamigration/jds/persistence/docstore/IDocumentDao.java:11]: issue: dao should always return the entity not the dto

File: src/main/java/com/datamigration/jds/persistence/param/DocumentParamDao.java [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+			IDocumentParamSQLs.CREATE_DOCUMENT_PARAMS_TABLE_SQL)) {
+			preparedStatement.execute();
+		} catch (SQLException e) {
+			throw new JPEPersistenceException(e, ErrorCode.DB_CREATE_ERROR);
+		}
+	}
+
+	@Override
+	public Optional<JivsDocumentParam> getById(UUID id) throws JPEPersistenceException {
+		return Optional.empty();
```
Comment (Involute1) [inline src/main/java/com/datamigration/jds/persistence/param/DocumentParamDao.java:35]: nitpick: why not implemented? if we dont need it throw an error pls

File: src/main/java/com/datamigration/jds/persistence/param/DocumentParamDao.java [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+	}
+
+	@Override
+	public Optional<List<JivsDocumentParam>> getAllAsList() throws JPEPersistenceException {
+		return Optional.empty();
```
Comment (Involute1) [inline src/main/java/com/datamigration/jds/persistence/param/DocumentParamDao.java:40]: nitpick: why not implemented? if we dont need it throw an error pls

File: src/main/java/com/datamigration/jds/persistence/param/DocumentParamDao.java [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+		if (!resultMap.isEmpty()) {
+			insertedJivsDocumentParam = new JivsDocumentParam(jivsDocumentParam.getDocumentId(), resultMap);
+		}
+
+		return insertedJivsDocumentParam;
+	}
+
+	@Override
+	public void update(JivsDocumentParam jivsDocumentParam) throws JPEPersistenceException {
+
```
Comment (Involute1) [inline src/main/java/com/datamigration/jds/persistence/param/DocumentParamDao.java:81]: nitpick: why not implemented? if we dont need it throw an error pls

File: src/main/java/com/datamigration/jds/persistence/param/IDocumentParamDao.java [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+	void createTables() throws JPEPersistenceException;
+
+	/**
+	 * Updates the params of the entity by the id in the database. Throws an exception if an error occurs.
+	 *
+	 * @param id     the id of the entity
+	 * @param params the params of the entity
+	 * @throws JPEPersistenceException if an error occurs during persisting
+	 */
+	void updateParams(UUID id, Map<String, String> params) throws JPEPersistenceException;
```
Comment (Involute1) [inline src/main/java/com/datamigration/jds/persistence/param/IDocumentParamDao.java:21]: nitpick: should also return the updated Map as optional

File: src/main/java/com/datamigration/jds/service/DocumentService.java [hunk 13]
```diff
@@ -0,0 +1,191 @@
...
+		DocumentDTO documentDTO = DTOUtil.toDocumentDTO(document);
+		DocumentDTO insertedDocumentDTO = documentDao.insert(documentDTO);
+
+		if (document.getParams() != null && !document.getParams().isEmpty()) {
+			JivsDocumentParam jivsDocumentParams = new JivsDocumentParam(insertedDocumentDTO.id(), document.getParams());
+			documentParamDao.insert(jivsDocumentParams);
+		}
+
+		document.setId(insertedDocumentDTO.id());
+		return DTOUtil.toDocumentDTO(document);
```
Comment (Involute1) [inline src/main/java/com/datamigration/jds/service/DocumentService.java:47]: nitpick: why not return insertedDocumentDto ?

File: src/main/java/com/datamigration/jds/util/DatabaseConfig.java
Comment (Involute1) [inline src/main/java/com/datamigration/jds/util/DatabaseConfig.java:1]: issue: this wont be need when you use jivs-config

File: src/main/java/com/datamigration/jds/util/exceptions/ErrorCode.java [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+	DB_CONNECT_ERROR("JPE-602", "Error connecting to the database"),
+	DB_WRITE_ERROR("JPE-603", "Error writing to the database"),
+	DB_EXECUTE_ERROR("JPE-604", "Error executing statement on the database"),
+	DB_CREATE_ERROR("JPE-605", "Error trying to create an object in the database"),
+	DB_CONFIG_ERROR("JPE-606", "Error regarding the libraries db configuration"),
+	DB_NO_RESULT_ERROR("JPE-607", "Error. No result was returned."),
+
+	SYSTEM_CONVERT_ERROR("JPE-701", "Error transforming DTOs to business objects"),
+	MISSING_EMPTY_CONSTRUCTOR_ERROR("JPE-702", "Error, empty constructor for business class is missing"),
+	WATCHER_CONFIG_ERROR("JPE-703", "Error regarding the libraries watcher configuration"),
```
Comment (Involute1) [inline src/main/java/com/datamigration/jds/util/exceptions/ErrorCode.java:21]: issue: the JPE is not correct since we are not in the process engine

File: src/main/java/com/datamigration/jds/util/exceptions/checked/JPEException.java
Comment (Involute1) [inline src/main/java/com/datamigration/jds/util/exceptions/checked/JPEException.java:1]: issue: class Name should use the correct abbreviation

File: src/main/java/com/datamigration/jds/util/exceptions/checked/JPENotImplementedException.java
Comment (Involute1) [inline src/main/java/com/datamigration/jds/util/exceptions/checked/JPENotImplementedException.java:1]: issue: class Name should use the correct abbreviation

File: src/main/java/com/datamigration/jds/util/exceptions/checked/JPEPersistenceException.java
Comment (Involute1) [inline src/main/java/com/datamigration/jds/util/exceptions/checked/JPEPersistenceException.java:1]: issue: class Name should use the correct abbreviation

File: src/main/java/com/datamigration/jds/util/exceptions/checked/JPESystemException.java
Comment (Involute1) [inline src/main/java/com/datamigration/jds/util/exceptions/checked/JPESystemException.java:1]: issue: class Name should use the correct abbreviation

File: src/main/java/com/datamigration/jds/util/exceptions/unchecked/JPEUncheckedConfigurationException.java
Comment (Involute1) [inline src/main/java/com/datamigration/jds/util/exceptions/unchecked/JPEUncheckedConfigurationException.java:1]: issue: class Name should use the correct abbreviation

File: src/main/java/com/datamigration/jds/util/exceptions/unchecked/JPEUncheckedException.java
Comment (Involute1) [inline src/main/java/com/datamigration/jds/util/exceptions/unchecked/JPEUncheckedException.java:1]: issue: class Name should use the correct abbreviation

File: src/main/java/com/datamigration/jds/util/exceptions/unchecked/JPEUncheckedNotImplementedException.java
Comment (Involute1) [inline src/main/java/com/datamigration/jds/util/exceptions/unchecked/JPEUncheckedNotImplementedException.java:1]: issue: class Name should use the correct abbreviation

File: src/main/java/com/datamigration/jds/util/exceptions/unchecked/JPEUncheckedPersistenceException.java
Comment (Involute1) [inline src/main/java/com/datamigration/jds/util/exceptions/unchecked/JPEUncheckedPersistenceException.java:1]: issue: class Name should use the correct abbreviation

File: src/main/java/com/datamigration/jds/model/docstore/JivsDocument.java [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+		this.creatorId = creatorId;
+		this.customerId = customerId;
+		this.systemId = systemId;
+		this.caseId = caseId;
+		this.params = params;
+		this.deleted = false;
+		this.created = LocalDateTime.now();
+	}
+
+	public JivsDocument(byte[] fileBin, String filename, String documentType, UUID creatorId, UUID customerId, UUID systemId,
```
Comment (mhn4jivs) [inline src/main/java/com/datamigration/jds/model/docstore/JivsDocument.java:35]: Question: Why do we have these two constructors? It does not make sense to me. I would make the deleted a default on DB side and not use it here

File: src/main/java/com/datamigration/jds/persistence/docstore/DocumentDao.java (hunk 7 above)
Comment (hdg4jivs) [reply src/main/java/com/datamigration/jds/persistence/docstore/DocumentDao.java:77]: Because as of task definition we only update the params

File: src/main/java/com/datamigration/jds/persistence/docstore/DocumentDao.java [hunk 16]
```diff
@@ -0,0 +1,298 @@
...
+	 * createTables - A method to create tables in the database.
+	 *
+	 * @throws JPEPersistenceException exception thrown if there is an issue with persistence
+	 */
+	@Override
+	public void createTables() throws JPEPersistenceException {
+		try (Connection connection = connect(); PreparedStatement preparedStatement = connection.prepareStatement(
+			IDocumentSQLs.CREATE_DOCSTORE_TABLE_SQL)) {
+			preparedStatement.execute();
+		} catch (SQLException e) {
```
Comment (mhn4jivs) [inline src/main/java/com/datamigration/jds/persistence/docstore/DocumentDao.java:37]: nitpick: logging

File: src/main/java/com/datamigration/jds/persistence/docstore/DocumentDao.java [hunk 17]
```diff
@@ -0,0 +1,298 @@
...
+			preparedStatement.setString(1, documentType);
+			try (ResultSet rs = preparedStatement.executeQuery()) {
+				List<DocumentDTO> list = new ArrayList<>();
+				while (rs.next()) {
+					DocumentDTO documentDTO = createDocumentDTO(rs);
+					list.add(documentDTO);
+				}
+				result = Optional.of(list);
+			}
+		} catch (Exception e) {
```
Comment (mhn4jivs) [inline src/main/java/com/datamigration/jds/persistence/docstore/DocumentDao.java:112]: nitpick: logging in whole class

File: src/main/java/com/datamigration/jds/persistence/docstore/DocumentDao.java [hunk 18]
```diff
@@ -0,0 +1,298 @@
...
+				throw new JPEPersistenceException(e, ErrorCode.DB_READ_ERROR);
+			}
+		} catch (SQLException e) {
+			throw new JPEPersistenceException(e, ErrorCode.DB_READ_ERROR);
+		}
+		return result;
+	}
+
+	@Override
+	public boolean updateDeleteFlag(UUID id) throws JPEPersistenceException {
```
Comment (mhn4jivs) [inline src/main/java/com/datamigration/jds/persistence/docstore/DocumentDao.java:260]: nitpick: naming? update to what?

File: src/main/java/com/datamigration/jds/persistence/docstore/IDocumentSQLs.java
Comment (mhn4jivs) [inline src/main/java/com/datamigration/jds/persistence/docstore/IDocumentSQLs.java:1]: praise: nice escaping

File: src/main/java/com/datamigration/jds/persistence/param/DocumentParamDao.java
Comment (mhn4jivs) [inline src/main/java/com/datamigration/jds/persistence/param/DocumentParamDao.java:1]: nitpick: logging

File: src/main/resources/config.properties
Comment (mhn4jivs) [inline src/main/resources/config.properties:1]: question: is this placed correctly?

**Review by mhn4jivs — CHANGES_REQUESTED** [review]

File: src/main/java/com/datamigration/jds/model/docstore/JivsDocument.java (hunk 15 above)
Comment (hdg4jivs) [reply src/main/java/com/datamigration/jds/model/docstore/JivsDocument.java]: Make sense, I wrote it because of the easier usage in test. Removing one constructor and the deleted param. 
Setting the deleted column to false on insert.

File: src/main/java/com/datamigration/jds/controller/DocStoreManager.java (hunk 2 above)
Comment (hdg4jivs) [reply src/main/java/com/datamigration/jds/controller/DocStoreManager.java:184]: Might be an unfortunate name of the parameter, I'm checking if the Optional<List<DocumentDTO>> contains a list or if it's null, I believe

Comment (hdg4jivs) [reply src/main/java/com/datamigration/jds/controller/DocStoreManager.java:184]: Actually you are right. In those getList methods in the dao, the Optional.of(new ArrayList()) is always assigned to the return value if there're no errors.

File: src/main/java/com/datamigration/jds/controller/DocStoreManager.java [hunk 19]
```diff
@@ -0,0 +1,189 @@
...
+	}
+
+	/**
+	 * Updates the documents params
+	 * @param id the id of the document
+	 * @param params the params map of the document
+	 * @return param map
+	 * @throws JDSPersistenceException if there is an error while establishing the database connection
+	 * */
+	public Map<String, String> updateParams(UUID id, Map<String, String> params) throws JDSPersistenceException {
```
Comment (mhn4jivs) [inline src/main/java/com/datamigration/jds/controller/DocStoreManager.java:152]: question: not sure what the decision was @Involute1 but did we not say that updates are void?

File: src/main/java/com/datamigration/jds/controller/DocStoreManager.java [hunk 20]
```diff
@@ -0,0 +1,189 @@
...
+		return documentWithParams;
+	}
+
+	/**
+	 * Gets the documents for the provided document type
+	 * @param documentType the type of the document
+	 * @return List of documents
+	 * @throws JDSPersistenceException if there is an error while establishing the database connection
+	 * */
+	public List<JivsDocument> getByDocumentType(String documentType) throws JDSPersistenceException {
```
Comment (mhn4jivs) [inline src/main/java/com/datamigration/jds/controller/DocStoreManager.java:75]: nitpick: if we get a list maybe we should describe it in the method name. I think there is a convention that we add "asList" to the method name.

File: src/main/java/com/datamigration/jds/persistence/DatabaseManager.java [hunk 21]
```diff
@@ -0,0 +1,101 @@
...
+			String dbPassword = config.getOptionalValue("jpe.mssql.password", String.class).orElse(null);
+			if (dbUrl != null && dbUsername != null && dbPassword != null) {
+				result = DriverManager.getConnection(dbUrl, dbUsername, dbPassword);
+				logger.debug("Connected to {}", result.getMetaData().getURL());
+			} else {
+				logger.error("{} : dbUrl={}, dbUser={}, pwdWasSet={}", ErrorCode.DB_CONFIG_ERROR.getTitle(),
+					dbUrl, dbUsername, dbPassword != null);
+				throw new JDSPersistenceException(ErrorCode.DB_CONFIG_ERROR);
+			}
+		} catch (SQLException e) {
```
Comment (mhn4jivs) [inline src/main/java/com/datamigration/jds/persistence/DatabaseManager.java:75]: nitpick: inconsistent exception handling here. also logging is weird. plus ou dont want to printout the password

File: src/main/java/com/datamigration/jds/persistence/db/DbDao.java [hunk 22]
```diff
@@ -0,0 +1,22 @@
...
+
+public class DbDao implements IDbDao {
+
+	@Override
+	public void createJDSSchema() throws JDSPersistenceException {
+		DatabaseManager databaseManager = DatabaseManager.getInstance();
+		try (Connection connection = databaseManager.connect(); PreparedStatement preparedStatement = connection.prepareStatement(
+			IDbSQLs.CREATE_JDS_SCHEMA_SQL)) {
+			preparedStatement.execute();
+		} catch (SQLException | JDSPersistenceException e) {
```
Comment (mhn4jivs) [inline src/main/java/com/datamigration/jds/persistence/db/DbDao.java:18]: question: y u no log

File: src/main/java/com/datamigration/jds/persistence/docstore/DocumentDao.java [hunk 23]
```diff
@@ -0,0 +1,310 @@
...
+			preparedStatement.setString(2, document.getFilename());
+			preparedStatement.setString(3, document.getDocumentType());
+			preparedStatement.setObject(4, document.getCreatorId());
+			preparedStatement.setObject(5, document.getCreatedAt());
+			preparedStatement.setObject(6, document.getCustomerId());
+			preparedStatement.setObject(7, document.getSystemId());
+			preparedStatement.setObject(8, document.getCaseId());
+			preparedStatement.setBoolean(9, false);
+			try (ResultSet rs = preparedStatement.executeQuery()) {
+				if (rs.next()) {
```
Comment (mhn4jivs) [inline src/main/java/com/datamigration/jds/persistence/docstore/DocumentDao.java:61]: nitpick: since you are catching the same exception, could you maybe merge the catches to the outer catch? Also if you use insert you could split it up into something like that as a general idea, you can discuss with @Involute1 :
```

int insertedRows = preparedStatement.executeUpdate();
if (insertedRows == 0) {
    logger.error("Document insertion failed.");
    throw new JDSPersistenceException(ErrorCode.DB_NO_RESULT_ERROR);
}

try (ResultSet rs = preparedStatement.getGeneratedKeys()) {
    if (rs.next()) {
        UUID id = UUID.fromString(rs.getString(1));
        document.setId(id);
        result = document;
    } else {
        logger.error("Document could not be inserted, no ID generated.");
        throw new JDSPersistenceException(ErrorCode.DB_NO_RESULT_ERROR);
    }
}
```

File: src/main/java/com/datamigration/jds/persistence/docstore/DocumentDao.java [hunk 24]
```diff
@@ -0,0 +1,310 @@
...
+			IDocumentSQLs.SELECT_DOCUMENT_BY_CASE_ID_SQL)) {
+			preparedStatement.setObject(1, id);
+			try (ResultSet rs = preparedStatement.executeQuery()) {
+				while (rs.next()) {
+					JivsDocument document = createDocument(rs);
+					result.add(document);
+				}
+			} catch (SQLException e) {
+				logger.error(e.getMessage(), e);
+				throw new JDSPersistenceException(e, ErrorCode.DB_READ_ERROR);
```
Comment (mhn4jivs) [inline src/main/java/com/datamigration/jds/persistence/docstore/DocumentDao.java:231]: nitpick: exception merging maybe too

File: src/main/java/com/datamigration/jds/persistence/docstore/DocumentDao.java [hunk 25]
```diff
@@ -0,0 +1,310 @@
...
+	public List<JivsDocument> getAllAsList() throws JDSPersistenceException {
+		List<JivsDocument> result = new ArrayList<>();
+		try (Connection connection =  DatabaseManager.getInstance().connect(); PreparedStatement preparedStatement = connection.prepareStatement(
+			IDocumentSQLs.SELECT_ALL_DOCUMENTS_SQL)) {
+			try (ResultSet rs = preparedStatement.executeQuery()) {
+				while (rs.next()) {
+					JivsDocument document = createDocument(rs);
+					result.add(document);
+				}
+			} catch (SQLException e) {
```
Comment (mhn4jivs) [inline src/main/java/com/datamigration/jds/persistence/docstore/DocumentDao.java:250]: nitpick: exception merging maybe too

File: src/main/java/com/datamigration/jds/persistence/docstore/IDocumentDao.java
Comment (mhn4jivs) [inline src/main/java/com/datamigration/jds/persistence/docstore/IDocumentDao.java:1]: nitpick: same thing with the naming "asList" i mentioned before

File: src/main/java/com/datamigration/jds/persistence/param/DocumentParamDao.java [hunk 26]
```diff
@@ -0,0 +1,148 @@
...
+			try (Connection connection = DatabaseManager
+				.getInstance().connect(); PreparedStatement preparedStatement = connection.prepareStatement(
+				IDocumentParamSQLs.INSERT_PARAMS_SQL, Statement.RETURN_GENERATED_KEYS)) {
+
+				preparedStatement.setObject(1, jivsDocumentParam.getDocumentId());
+				preparedStatement.setString(2, entry.getKey());
+				preparedStatement.setString(3, entry.getValue());
+				try (ResultSet rs = preparedStatement.executeQuery()) {
+					if (rs.next()) {
+						resultMap.put(entry.getKey(), entry.getValue());
```
Comment (mhn4jivs) [inline src/main/java/com/datamigration/jds/persistence/param/DocumentParamDao.java:63]: nitpick: exception merging maybe too

File: src/main/java/com/datamigration/jds/service/DocumentService.java
Comment (mhn4jivs) [inline src/main/java/com/datamigration/jds/service/DocumentService.java:1]: nitpick: naming with asList

**Review by mhn4jivs — CHANGES_REQUESTED** [review]

File: src/main/java/com/datamigration/jds/controller/DocStoreManager.java (hunk 19 above)
Comment (Involute1) [reply src/main/java/com/datamigration/jds/controller/DocStoreManager.java:152]: yes this is true but i forgot to tell @hdg4jivs

File: src/main/java/com/datamigration/jds/persistence/docstore/DocumentDao.java (hunk 23 above)
Comment (Involute1) [reply src/main/java/com/datamigration/jds/persistence/docstore/DocumentDao.java:61]: generally @mhn4jivs approach is a bit cleaner, but i am fine with both. But in any case i would try to merge the catch blocks since you are catching the same thing over and over again

Comment (hdg4jivs) [reply src/main/java/com/datamigration/jds/persistence/docstore/DocumentDao.java:61]: Not possible without an actual identity, which has to be of a numeric type, column.
